packages/jaeger-ui/src/model/ddg/PathElem.test.js

```diff
@@ -52,4 +54,58 @@ describe('PathElem', () => {
       pathElem.visibilityIdx = testVisibilityIdx;
     }).toThrowError();
   });
+
+  it('has focalSideNeighbor if distance is not 0', () => {
+    expect(pathElem.focalSideNeighbor).toBe(testPath.members[testMemberIdx + 1]);
+  });
+
+  it('has a null focalSideNeighbor if distance is 0', () => {
+    pathElem = new PathElem({ path: testPath, operation: testOperation, memberIdx: testPath.focalIdx });
+    expect(pathElem.focalSideNeighbor).toBe(null);
+  });
+
+  it('has correct farSideEdgesKey and focalSideEdgesKey when upstream', () => {
+    expect(pathElem.farSideEdgesKey).toBe('ingressEdges');
+    expect(pathElem.focalSideEdgesKey).toBe('egressEdges');
+  });
+
+  it('has correct farSideEdgesKey and focalSideEdgesKey when downstream', () => {
+    pathElem = new PathElem({ path: testPath, operation: testOperation, memberIdx: testPath.focalIdx + 1 });
+    expect(pathElem.farSideEdgesKey).toBe('egressEdges');
+    expect(pathElem.focalSideEdgesKey).toBe('ingressEdges');
+  });
+
+  describe('legibility', () => {
+    const operations = simplePath.map(({ operation, service }) => ({
+      name: operation,
+      service: {
+        name: service,
+      },
+    }));
+    const path = {
+      focalIdx: 2,
+    };
+    const members = operations.map((operation, i) => new PathElem({ memberIdx: i, operation, path }));
```

Comment:
`members` can be made directly from `simplePath.map` rather than needing to `map` twice.